packages/fileeditor-extension/src/index.ts

```diff
@@ -316,11 +316,14 @@ function activate(app: JupyterLab, restorer: ILayoutRestorer, editorServices: IE
             break;
           }
         }
-      } else {
+        selected = blocks.length > 0;
+      }
+
+      if (!selected) {
         // no selection, submit whole line and advance
         code = editor.getLine(selection.start.line);
         const cursor = editor.getCursorPosition();
-        if (cursor.line + 1 == editor.lineCount) {
+        if (cursor.line + 1 === editor.lineCount) {
           let text = editor.model.value.text;
           editor.model.value.text = text + '\n';
         }
```

Comment:
This resetting of the cursor position is not part of this particular PR, but I found it surprising, and did not particularly like it.

Reply:
The cursor change allows you to step through a python file by repeatedly pressing Shift+Enter.  Care to open an issue for discussion?

Reply:
Sure, it's not really relevant here.